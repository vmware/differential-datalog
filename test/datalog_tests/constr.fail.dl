// constructor names have to uppercase
typedef t = lowercase { field: bigint }

//---

// constructor names have to be unique
typedef t = C { f: bigint } |
            C { f: string }

//---

// Union types must be specified by name
function(x : Constructor{}): bigint

//---

// nested constructors must use names
typedef t19 = Const { f0: Const1 { f2: bigint } }

//---

// function names have to be lowercase
function F(): bigint

//---

// Constructor argument types must be unambiguous
typedef Alt = C0{x: bit<16>}
            | C1{x: bit<32>}

//---

// type argument 'D not used in type definition
typedef Parameterized<'A,'B,'C,'D> = Option1{x: 'A}
                                   | Option2{y: 'B, z: 'C}
