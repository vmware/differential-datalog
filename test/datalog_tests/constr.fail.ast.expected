Failed to parse input file: "./test/datalog_tests/constr.fail.dl" (line 2, column 23):
unexpected '{'
expecting "import", "typedef", "extern", "input", "output", "relation", "function", relation name, "for" or end of input

error: ./test/datalog_tests/constr.fail.dl:4:13-4:29: Multiple definitions of constructor C at the following locations:
  ./test/datalog_tests/constr.fail.dl:4:13-4:29
  ./test/datalog_tests/constr.fail.dl:5:13-7:1

Failed to parse input file: "./test/datalog_tests/constr.fail.dl" (line 4, column 9):
unexpected "("
expecting end of "function" or function name

Failed to parse input file: "./test/datalog_tests/constr.fail.dl" (line 4, column 34):
unexpected "{"
expecting ".", "<", "," or "}"

Failed to parse input file: "./test/datalog_tests/constr.fail.dl" (line 4, column 10):
unexpected F
expecting function name

error: ./test/datalog_tests/constr.fail.dl:4:15-7:1: Field x is declared with different types

error: ./test/datalog_tests/constr.fail.dl:4:1-6:1: The following type variables are not used in type definition: D
